spec/mauth_client_spec.rb

```diff
@@ -52,237 +66,523 @@ def (::Rails).logger
+                  message
+                )
+            end
+          end
+        end
+
+        it "considers an authentically-signed request to be inauthentic when it has no x-mws-time" do
+          v1_signed_req.headers.delete('X-MWS-Time')
+          expect { authenticating_mc.authenticate!(v1_signed_req) }.to raise_error(
+              MAuth::InauthenticError,
+              /Time verification failed\. No x-mws-time present\./
+            )
+        end
+
+        it "considers a request with no X-MWS-Authentication to be inauthentic" do
+          v1_signed_req.headers.delete('X-MWS-Authentication')
+          expect { authenticating_mc.authenticate!(v1_signed_req) }.to raise_error(
+              MAuth::MauthNotPresent,
+              'Authentication Failed. No mAuth signature present;  X-MWS-Authentication header is blank, MCC-Authentication header is blank.'
+            )
+        end
+
+        it "considers a request with a bad MWS token to be inauthentic" do
+          ['mws', 'm.w.s', 'm w s', 'NWS', ' MWS'].each do |bad_token|
+              v1_signed_req.headers['X-MWS-Authentication'] = v1_signed_req.headers['X-MWS-Authentication'].sub(/\AMWS/, bad_token)
+              expect { authenticating_mc.authenticate!(v1_signed_req) }.to raise_error(
+                MAuth::InauthenticError, /Token verification failed\. Expected MWS; token was .*/)
+          end
+        end
+
+        [::Faraday::Error::ConnectionFailed, ::Faraday::Error::TimeoutError].each do |error_klass|
+          it "raises UnableToAuthenticate if mauth is unreachable with #{error_klass.name}" do
+            allow(test_faraday).to receive(:get).and_raise(error_klass.new(''))
+            allow(test_faraday).to receive(:post).and_raise(error_klass.new(''))
+              expect { authenticating_mc.authentic?(v1_signed_req) }.to raise_error(MAuth::UnableToAuthenticateError)
+          end
+        end
+
+        it "raises UnableToAuthenticate if mauth errors" do
+          stubs.instance_eval{ @stack.clear } #HAX
+          stubs.get("/mauth/v1/security_tokens/#{app_uuid}.json") { [500, {}, []] } # for the local authenticator
+          stubs.post('/mauth/v1/authentication_tickets.json') { [500, {}, []] } # for the remote authenticator
+          expect {  authenticating_mc.authentic?(v1_signed_req) }.to raise_error(MAuth::UnableToAuthenticateError)
+        end
+
+        describe 'logging requester and requestee' do
+          before do
+            allow(authenticating_mc).to receive(:client_app_uuid).and_return('authenticator')
+          end
+
+          it 'logs the mauth app uuid of the requester and requestee when they both have such uuids' do
+            v1_signed_req = client.signed(request, time: Time.now.to_i)
+            expect(authenticating_mc.logger).to receive(:info).with("Mauth-client attempting to authenticate request from app with mauth app uuid signer to app with mauth app uuid authenticator using version MWSV2.")
```

Comment:
Thanks for catching, these two specs were actually signing with both v2 and v1 because they were overwriting the `let` `v1_signed_req` with their own local `v1_signed_req` that was being called without the override. Fixed and confirmed no other specs are doing this.